src/plugins/procdump/procdump.h

```diff
@@ -117,6 +117,7 @@ struct procdump_config
 {
     const char* procdump_dir;
     bool compress_procdumps;
+    std::map<vmi_pid_t, bool>& terminated_processes;
```

Comment:
`std::unordered_map` is better, as we don't need the automatic ordering (naturally this should be propagated further).

Reply:
again, This is a non-owning non-nullable pointer, which is stored. I'm **strongly** against that [...] please propagate this change across this PR.

Reply:
Have switched to `std::shared_ptr`

Reply:
Switchd to `unordered_map`